kahuna/public/js/search/query.js

```diff
@@ -62,11 +56,13 @@ query.controller('SearchQueryCtrl',
     Object.keys($stateParams)
           .forEach(setAndWatchParam);
 
-    // pass undefined to the state on empty to remove the QueryString
-    function valOrUndefined(str) { return str || undefined; }
+    // URL parameters are not decoded when taken out of the params.
+    // Might be fixed with: https://github.com/angular-ui/ui-router/issues/1759
+    // Pass undefined to the state on empty to remove the QueryString
+    function valOrUndefined(str) { return str ? decodeURIComponent(str) : undefined; }
```

Comment:
Just to check that we do want to `decodeURIComponent` both when `ctrl.filter[key]` is initialised and updated below? (does seem like both could be unified, but this it OT here!)